pkg/validator/csv.go

```diff
@@ -0,0 +1,32 @@
+package validator
+
+import (
+	"bytes"
+	"io"
+
+	"encoding/csv"
+)
+
+// CsvValidator is used to validate a byte slice that is intended to represent a
+// CSV file.
+type CsvValidator struct{}
+
+// Validate checks if the provided byte slice represents a valid .csv file.
+// https://pkg.go.dev/encoding/csv
+func (csvv CsvValidator) Validate(b []byte) (bool, error) {
+	csvReader := csv.NewReader(bytes.NewReader(b))
```

Comment:
Does this implementation support multiple delimiters or just commas?

Reply:
You can customize the delimiters using an option. Do we want to support the same? if so how do we pass this custom delimiter?

Reply:
Is it possible to do an initial parsing of the file to determine the value of the delimiter? There is no good way to pass custom configurations to validators currently. Even if we could there could be different delimiters for different CSV files in the scan

Reply:
Let me see but there is no one way to do that. If one were to use different de-limiters in each line it could be confusing.

Reply:
If it's not fairly straightforward to implement I'm good with only supporting commas and documenting that for this MR and creating an issue to look at supporting multiple delimiter in a future story

Reply:
@kehoecj Currently it supports a non standard delimiter as well as in the following is also a valid csv

```
first_name:last_name:username
Rob:Pike:rob
Ken:Thompson:ken
Robert:Griesemer:gri
```

Reply:
Basically it identifies the non standard delimiter and accepts it. It rejects mismatches if any.

Reply:
I'll create a follow-up issue